packages/react/src/hooks/useOverflow.ts

```diff
@@ -8,20 +8,51 @@ export function useOverflow<T extends HTMLElement>(ref: React.RefObject<T>) {
       return
     }
 
-    const observer = new ResizeObserver(entries => {
+    // Track whether this is the first callback (fires immediately on observe())
+    let isFirstCallback = true
+    let pendingFrame: number | null = null
+    let latestEntries: ResizeObserverEntry[] | null = null
+
+    const checkOverflow = (entries: ResizeObserverEntry[]) => {
       for (const entry of entries) {
         if (
           entry.target.scrollHeight > entry.target.clientHeight ||
           entry.target.scrollWidth > entry.target.clientWidth
         ) {
           setHasOverflow(true)
-          break
+          return
         }
       }
+      setHasOverflow(false)
```

Comment:
I believe this is a current bug, since things can never go to `not overflow` states